angular/src/di/r3_injector.ts

```diff
@@ -30,5 +30,5 @@ export abstract class EnvironmentInjector implements Injector {
   /**
    * @internal
    */
-  abstract onDestroy(callback: () => void): void;
+  abstract onDestroy?(callback: () => void): void;
```

Comment:
Why was this change needed?

Reply:
This was the align with the type signature of the `EnvironmentInjector` type from `@angular/core`. We are maintaining our own "version" of the types, since we cannot update our library dependencies to target Angular 14. When I originally ported this, I forgot to make the `onDestroy` callback optional.

Without this change, developers would receive a typescript language server warning in their IDE that the type of their `environmentInjector` does not match the type signature of what we are expecting to be passed in.